modules/statics/src/account.ts

```diff
@@ -95,12 +95,36 @@ export class ContractAddressDefinedToken extends AccountCoinToken {
   }
 }
 
+/**
+ * ERC20 token addresses are Base58 formatted on some blockchains.
+ */
+export class Base58ContractAddressDefinedToken extends AccountCoinToken {
+  public contractAddress: ContractAddress;
+
+  constructor(options: Erc20ConstructorOptions) {
+    super({
+      ...options,
+    });
+
+    if (!/^[1-9A-HJ-NP-Za-km-z]{34}$/.test(options.contractAddress)) {
+      throw new InvalidContractAddressError(options.name, options.contractAddress);
+    }
+
+    this.contractAddress = (options.contractAddress as unknown) as ContractAddress;
+  }
+}
```

Comment:
Which are the use cases for this? why are we creating a new class where the only difference is the regex used to validate the address?
Maybe this could be model different, depending on the usage for each one

Reply:
yeah this doesn't seem great. i think this anti-pattern was setup for other things since there's a weird hierarchy to these. i'm not terribly against it since it's already kind of in there as an anti pattern